src/ex1/script.js

```diff
@@ -0,0 +1,141 @@
+  if (event.key === "Enter") {
+    const inputElem = document.querySelector(".add-task-input");
+    if (inputElem === document.activeElement) {
+      onAddTask(event);
+    }
+  }
+});
+
+const createTask = () => {
+  const task = document.createElement("div");
+  task.classList.add("task-container");
+  task.id = `task-${taskId}`;
+  task.innerHTML = `
+    <div class="task-checkbox">
+        <input type="checkbox" id="input-${taskId}" />
+        <label for="input-${taskId}" />
+    </div>
+    <div class="task-txt-container">
+        <p class="task-txt" id="task-txt-${taskId}" >${addTaskInput.value}</p>
+    </div>
+    <button id="trash-button-${taskId}" class="trash">
+        <svg id="trash-svg-${taskId}" fill="#000000" xmlns="http://www.w3.org/2000/svg"  viewBox="0 0 26 26" width="20px" height="20px">
+            <path id="trash-path-${taskId}" d="M 11 -0.03125 C 10.164063 -0.03125 9.34375 0.132813 8.75 0.71875 C 8.15625 1.304688 7.96875 2.136719 7.96875 3 L 4 3 C 3.449219 3 3 3.449219 3 4 L 2 4 L 2 6 L 24 6 L 24 4 L 23 4 C 23 3.449219 22.550781 3 22 3 L 18.03125 3 C 18.03125 2.136719 17.84375 1.304688 17.25 0.71875 C 16.65625 0.132813 15.835938 -0.03125 15 -0.03125 Z M 11 2.03125 L 15 2.03125 C 15.546875 2.03125 15.71875 2.160156 15.78125 2.21875 C 15.84375 2.277344 15.96875 2.441406 15.96875 3 L 10.03125 3 C 10.03125 2.441406 10.15625 2.277344 10.21875 2.21875 C 10.28125 2.160156 10.453125 2.03125 11 2.03125 Z M 4 7 L 4 23 C 4 24.652344 5.347656 26 7 26 L 19 26 C 20.652344 26 22 24.652344 22 23 L 22 7 Z M 8 10 L 10 10 L 10 22 L 8 22 Z M 12 10 L 14 10 L 14 22 L 12 22 Z M 16 10 L 18 10 L 18 22 L 16 22 Z"/>
+        </svg>
+    </button>
+    `;
+  taskId++;
+  return task;
+};
+
+const onAddTask = (event) => {
+  // Validate the input isn't empty before adding the task
+  if (!addTaskInput.value) {
+    return;
+  }
+
+  document.querySelector(".tasks-container").appendChild(createTask());
+
+  // Clear 'Add task' input
+  addTaskInput.value = "";
+  addTaskBtn.classList.remove("valid-task-btn");
+  isAddTaskClickable = false;
+
+  tasksCnt++;
+  updateTasksLeft();
+
+  if (tasksCnt === 1) {
+    // Hide finished all
+    document
+      .querySelector(".finished-all-missions")
+      .classList.toggle("finished-all-missions-active");
+    // Show clear all and left tasks
+    document.querySelector(".todo-footer-container").classList.toggle("hide");
+  }
+
+  const checkboxBtn = document.querySelector(`#input-${taskId - 1}`);
+  const taskTxt = document.querySelector(`#task-txt-${taskId - 1}`);
+  checkboxBtn.addEventListener("click", (event) => onChecked(event, taskTxt));
+
+  const input = document.querySelector(`#task-${taskId - 1}`);
+  input.addEventListener("click", (event) => onTaskPress(event));
+
+  const trashBtn = document.querySelector(`#trash-button-${taskId - 1}`);
+  trashBtn.addEventListener("click", (event) => onTaskDelete(event));
+};
+
+const onChecked = (event, taskTxt) => {
```

Comment:
would rename `taskTxt` param to `taskElement`, as it's more understood that you can apply DOM method on it

Reply:
Agree. Changed it in Ex2 to taskTxtElem. It was important for me that it will be clear that it is the task's text, since the next line is toggling a class that make only the text (and not the all div) crossed. I also added a comment that clarifies that: "// Crossing/uncrossing a done task's text"